conda_build/metadata.py

```diff
@@ -1706,6 +1725,7 @@ def get_output_metadata_set(self, permit_undefined_jinja=False,
             outputs = get_output_dicts_from_metadata(self)[0]
             output_tuples = [(outputs, self)]
         else:
+            all_output_metadata = OrderedDict()
```

Comment:
This is new - we used to only have the whole set of metadata during finalization.  With parsing whole recipes, though, we need to have that information earlier.  This was necessary for the reparsing of the outputs section.